hal/inc/power_hal.h

```diff
@@ -61,8 +61,9 @@ typedef struct hal_power_config {
     uint16_t charge_current; // charge current
     uint16_t termination_voltage; // termination voltage
     uint8_t soc_bits; // bits precision for SoC calculation (18 (default) or 19)
-    uint8_t reserved2;
-    uint32_t reserved3[4];
+    uint8_t aux_pwr_ctrl_pin; // pin number for auxiliary power control
+    uint8_t aux_pwr_ctrl_pin_level; // active level for auxiliary power control
+    uint32_t reserved3[3];
 } hal_power_config;
 static_assert(sizeof(hal_power_config) == 32, "hal_power_config size changed");
```

Comment:
To be clear, we are using `uint8_t reserved2` as well as one item of the `uint32_t reserved3[]` array to store the new pin and pin state?

I am assuming since we dont pack this structure that this static assert still passes? There should be any issues reading this out of DCT?

Reply:
Good catch. we should only consume one byte of the `reserved3` field.